python/map.c

```diff
@@ -254,16 +275,21 @@ static PyObject* PyUpb_MapContainer_Repr(PyObject* _self) {
   upb_map* map = PyUpb_MapContainer_GetIfWritable(self);
   PyObject* dict = PyDict_New();
   if (map) {
-    size_t iter = UPB_MAP_BEGIN;
     const upb_fielddef* f = PyUpb_MapContainer_GetField(self);
     const upb_msgdef* entry_m = upb_fielddef_msgsubdef(f);
     const upb_fielddef* key_f = upb_msgdef_field(entry_m, 0);
     const upb_fielddef* val_f = upb_msgdef_field(entry_m, 1);
+    size_t iter = UPB_MAP_BEGIN;
     while (upb_mapiter_next(map, &iter)) {
       PyObject* key =
           PyUpb_UpbToPy(upb_mapiter_key(map, iter), key_f, self->arena);
       PyObject* val =
           PyUpb_UpbToPy(upb_mapiter_value(map, iter), val_f, self->arena);
+      if (!key || !val) {
+        Py_XDECREF(key);
+        Py_XDECREF(val);
+        return NULL;
```

Comment:
I think you need to `PY_DECREF(dict)` as well

Reply:
Done.